financial-connections/src/main/java/com/stripe/android/financialconnections/domain/ConfirmVerification.kt

```diff
@@ -13,21 +14,64 @@ internal class ConfirmVerification @Inject constructor(
         consumerSessionClientSecret: String,
         verificationCode: String,
     ): ConsumerSession = requireNotNull(
-        consumerSessionRepository.confirmConsumerVerification(
-            consumerSessionClientSecret = consumerSessionClientSecret,
-            verificationCode = verificationCode,
-            type = VerificationType.SMS
+        kotlin.runCatching {
+            consumerSessionRepository.confirmConsumerVerification(
+                consumerSessionClientSecret = consumerSessionClientSecret,
+                verificationCode = verificationCode,
+                type = VerificationType.SMS
+            )
+        }.fold(
+            onSuccess = { it },
+            onFailure = { throw it.toDomainException(VerificationType.SMS) }
         )
     )
 
     suspend fun email(
         consumerSessionClientSecret: String,
         verificationCode: String,
     ): ConsumerSession = requireNotNull(
-        consumerSessionRepository.confirmConsumerVerification(
-            consumerSessionClientSecret = consumerSessionClientSecret,
-            verificationCode = verificationCode,
-            type = VerificationType.EMAIL
+        kotlin.runCatching {
+            consumerSessionRepository.confirmConsumerVerification(
+                consumerSessionClientSecret = consumerSessionClientSecret,
+                verificationCode = verificationCode,
+                type = VerificationType.EMAIL
+            )
+        }.fold(
+            onSuccess = { it },
+            onFailure = { throw it.toDomainException(VerificationType.EMAIL) }
         )
+
     )
+
+    /**
+     * Convert Exception to [OTPError] (a soft error related to OTP verification) or
+     * returns the original exception.
+     */
+    private fun Throwable.toDomainException(verificationCode: VerificationType): Throwable {
+        return when (val code = (this as? StripeException)?.stripeError?.code ?: "") {
```

Comment:
nit: can probably use `=` directly in method signature

Reply:
https://github.com/stripe/stripe-android/pull/6570/commits/eee474f512287aa031e4ed7f0deeb086f9d2ce24